compiler/qsc/src/interpret.rs

```diff
@@ -639,6 +672,45 @@ impl Interpreter {
         })
     }
 
+    /// Performs QIR codegen using the given callable with the given arguments on a new instance of the environment
+    /// and simulator but using the current compilation.
+    pub fn qirgen_from_callable(
+        &mut self,
+        callable: &Value,
+        args: Value,
+    ) -> std::result::Result<String, Vec<Error>> {
+        if self.capabilities == TargetCapabilityFlags::all() {
+            return Err(vec![Error::UnsupportedRuntimeCapabilities]);
+        }
+
+        let Value::Global(store_item_id, _) = callable else {
+            panic!("value is not a global callable");
+        };
+
+        fir_to_qir_from_callable(
+            &self.fir_store,
+            self.capabilities,
+            None,
+            *store_item_id,
+            args,
+        )
+        .map_err(|e| {
+            let hir_package_id = match e.span() {
+                Some(span) => span.package,
+                None => map_fir_package_to_hir(self.package),
+            };
+            let source_package = self
+                .compiler
+                .package_store()
+                .get(hir_package_id)
+                .expect("package should exist in the package store");
```

Comment:
Pretty sure I've done this in a bunch of places too, but perhaps these panics could be wrapped up in a new `Error::Internal(String)` or `Error::InvariantFailed(String)` for more robust code?

Reply:
I would say for this one it's ok, because getting all the way here and then not finding something in a package store is a terminal failure, not really something we'd want to recover from.